client/src/nv_ingest_client/message_clients/simple/simple_client.py

```diff
@@ -0,0 +1,243 @@
+        self._port = port
+        self._db = db
+        self._max_retries = max_retries
+        self._max_backoff = max_backoff
+        self._max_pool_size = max_pool_size
+        self._connection_timeout = connection_timeout
+        self._use_ssl = use_ssl
+
+    def get_client(self):
+        return self
+
+    def submit_message(self, queue_name: str, message: str, timeout: Optional[float] = None,
+                       for_nv_ingest: bool = False) -> ResponseSchema:
+        return self._handle_push(queue_name, message, timeout, for_nv_ingest)
+
+    def fetch_message(self, queue_name: str, timeout: Optional[float] = None) -> ResponseSchema:
+        return self._handle_pop(queue_name, timeout)
+
+    def ping(self) -> ResponseSchema:
+        command = {"command": "PING"}
+        return self._execute_simple_command(command)
+
+    def size(self, queue_name: str) -> ResponseSchema:
+        """Fetches the current number of items in the specified queue."""
+        command = {"command": "SIZE", "queue_name": queue_name}
+        return self._execute_simple_command(command)
+
+    def _handle_push(self, queue_name: str, message: str, timeout: Optional[float],
+                     for_nv_ingest: bool) -> ResponseSchema:
+        """Push a message to the queue, respecting the specified timeout."""
+        if (not queue_name or not isinstance(queue_name, str)):
+            return ResponseSchema(response_code=1, response_reason="Invalid queue name.")
+        if (not message or not isinstance(message, str)):
+            return ResponseSchema(response_code=1, response_reason="Invalid message.")
+
+        if (for_nv_ingest):
+            command = {"command": "PUSH_FOR_NV_INGEST", "queue_name": queue_name, "message": message}
+        else:
+            command = {"command": "PUSH", "queue_name": queue_name, "message": message}
+
+        if (timeout is not None):
+            command["timeout"] = timeout
+
+        start_time = time.time()
+        while True:
+            elapsed = time.time() - start_time
+            remaining_timeout = (timeout - elapsed) if (timeout is not None) else None
+            if ((remaining_timeout is not None) and (remaining_timeout <= 0)):
+                return ResponseSchema(response_code=1, response_reason="PUSH operation timed out.")
+
+            try:
+                with socket.create_connection((self._host, self._port), timeout=self._connection_timeout) as sock:
+                    self._send(sock, json.dumps(command).encode('utf-8'))
+                    # Receive initial response with transaction ID
+                    response_data = self._recv(sock)
+                    response = json.loads(response_data)
+
+                    if (response.get('response_code') != 0):
+                        if (response.get('response_reason') == "Queue is full" or
+                                response.get('response_reason') == "Queue is not available"):
+                            time.sleep(0.5)
```

Comment:
Is the idea here that the memory backing the queue is full and that would be the cause of a full queue here or is there an imposed maximum size?

Reply:
There is an imposed maximum size, per queue. Currently this defaults to 10,000 where its used.